version: "1.0" # mistral version

description: Will archive fastq files and SAV files

input:
  - archive_data
  - archive_settings
  - processing_api_service_url
  - processing_api_sequence_run_archive_status_url
  - mail_bioinfo

vars:
  - year: null
  - stderr: null

tasks:
  get_archive_year:
    action: core.local
    input:
      cmd: echo <% ctx(archive_data).date %> | awk 'BEGIN{FS="-"}{print($1)}'
    next:
      - when: <% succeeded() %>
        publish: 
         - year: <% result().stdout %>
        do:
         - archive_sequence_files

  archive_sequence_files:
    with:
      items: <% ctx(archive_data).analysis %>
    action: ductus.archive_sequence_files
    input:
      analysis_id: <% item() %>
      year: <% ctx(year) %>
      
    # next:
    #   - when: <% failed() %>
    #     do:
    #       - bioinfo_error_notifier


  # create_archive_done:
  #   join: all
  #   action: core.local
  #   input:
  #     cwd: <% ctx(runfolder_path) %>
  #     cmd: echo "<% ctx(runfolder_name) %>" >> archived_fastq_<% ctx(experiment_name) %>.txt
  #   next:
  #     - when: <% failed() %>
  #       publish:
  #         - stderr: <% result().stderr %>
  #         - failed_step: "create_archive_don  -- Could not add entries to <% ctx(runfolder_path) %>/archived_<% ctx(experiment_name) %>.txt"
  #       do:
  #         - bioinfo_error_notifier

  # bioinfo_error_notifier:
  #   action: core.sendmail
  #   input:
  #     to: <% ctx(mail_bioinfo) %>
  #     from: "stanley@clinicalgenomics-as.se"
  #     subject: "'[DUCTUS][WP][ERROR] - Archiving, <% ctx(runfolder_name) %>'"
  #     body: Something went wrong during the archiving of <% ctx(runfolder_name) %>, please investigate!!!\n Failure message -- <% ctx(failed_step) %>, <% ctx(stderr)% >
  #   next:
  #     - when: <% succeeded() %>
  #       do:
  #         - fail

output:
  - stderr: <% ctx(stderr) %>
